torch/_decomp/decompositions.py

```diff
@@ -728,23 +728,37 @@ def slice_forward(
     start_val = start if start is not None else 0
     end_val = end if end is not None else sys.maxsize  # 2^63 - 1
 
-    if guard_size_oblivious(start_val < 0):
+    # negative indexing case
+    if guard_or_false(start_val < 0):
```

Comment:
can we write this as .
the checks bellow are checking different things under different conditions 
one checks start_val when its unbacked 
and checks start_val +sizes[dim] when start_val is for example -10 and sizes[dim]  is unbacked. 
````
if guard_or_false(start_val < 0):
    start_val += sizes[dim]
    # sizes[dim] is always positive 
    torch._check(start_val >= 0, "the off-setted start_val {start_val} is assumed to be positive")
else:
   torch._check(start_val >= 0, "start_val {start_val} is assumed to be positive")
   ```

Reply:
same for end